orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/appengine/AppEngineBranchFinder.groovy

```diff
@@ -22,21 +22,66 @@ import com.netflix.spinnaker.orca.pipeline.model.Stage
 import java.util.regex.Pattern
 
 class AppEngineBranchFinder {
-  static void populateFromStage(Map operation, Stage stage) {
+  static String findInStage(Map operation, Stage stage) {
     if (operation.fromTrigger && operation.trigger && stage.execution instanceof Pipeline) {
       Map trigger = (stage.execution as Pipeline).trigger
-      def matchConditions = [
-        trigger?.source == operation.trigger.source,
-        trigger?.project == operation.trigger.project,
-        trigger?.slug == operation.trigger.slug,
-        trigger?.branch ==~ Pattern.compile(operation.trigger.branch as String)
-      ]
-
-      if (matchConditions.every()) {
-        operation.branch = trigger.branch
+
+      if (trigger.type == "git") {
+        return fromGitTrigger(operation, trigger)
+      } else if (trigger.type == "jenkins" || (trigger.type == "manual" && trigger.master && trigger.job)) {
+        return fromJenkinsTrigger(operation, trigger)
       } else {
-        throw new IllegalStateException("No branch found for repository ${operation.repositoryUrl} in trigger context.")
+        throw new IllegalArgumentException("Trigger type \'${trigger.type}\' not supported " +
```

Comment:
I don't think you need to escape the single-quotes, since the string itself uses double-quotes.